Spanner/tests/System/AdminTest.php

```diff
@@ -129,6 +130,50 @@ public function testDatabase()
         $this->assertEquals($db->ddl()[0], $stmt);
     }
 
+    public function testDatabaseDropProtection()
+    {
+        $instance = self::$instance;
+
+        $dbName = uniqid(self::TESTING_PREFIX);
+        $op = $instance->createDatabase($dbName);
+
+        $this->assertInstanceOf(LongRunningOperation::class, $op);
+        $db = $op->pollUntilComplete();
+        $this->assertInstanceOf(Database::class, $db);
+
+        $info = $db->reload();
+        $this->assertFalse($info['enableDropProtection']);
+
+        $op = $db->updateDatabase(['enableDropProtection' => true]);
+        $op->pollUntilComplete();
+        $info = $db->reload();
+        $this->assertTrue($info['enableDropProtection']);
+
+        // delete database should throw
```

Comment:
We could also try deleting instance and check if its deletion is blocked.

Reply:
Correct, but an instance might have multiple DBs to influence the result of delete.
We don't really create/delete a spanner instance in tests so i just tested on the db delete. More concrete this way.

Reply:
> Correct, but an instance might have multiple DBs to influence the result of delete.

But having drop protection enabled on one DB can influence the result of delete (instance deletion will be blocked).

> We don't really create/delete a spanner instance in tests so i just tested on the db delete. More concrete this way.

It makes sense for other tests not to delete a spanner instance. But this feature explicitly blocks deletion of an instance if at least one of the underlying databases has drop protection enabled. Hence, the requirement. This has been documented in the client spec doc.